official/keras_application_models/benchmark_main.py

```diff
@@ -0,0 +1,242 @@
+      samples in one epoch.
+    metrics: `dict` that maps string-valued tags to metric names, or `iterable`
+      of metric names.
+    metric_logger: instance of `BenchmarkLogger`, the benchmark logger that
+      callback should use to write the log.
+
+  Returns:
+    callbacks: a list of callbacks.
+
+  Raises:
+    ValueError: if an unrecognized callback name is passed.
+  """
+  if not callbacks_list:
+    return []
+
+  batch_based = True if benchmark_level == "batch_based" else None
+  epoch_based = True if benchmark_level == "epoch_based" else None
+
+  callbacks = []
+  for callback in callbacks_list:
+    callback_name = callback.strip().lower()
+    if callback_name not in model_callbacks.CALLBACKS:
+      raise ValueError("Unrecognized callback requested: {}".format(callback))
+
+    if callback_name == model_callbacks.CALLBACKS[0]:
+      exp_per_second_callback = model_callbacks.ExamplesPerSecondCallback(
+          batch_size=batch_size, epoch_size=epoch_size,
+          batch_based=batch_based, epoch_based=epoch_based,
+          metric_logger=metric_logger)
+      callbacks.append(exp_per_second_callback)
+    else:
+      logging_metric_callback = model_callbacks.LoggingMetricCallback(
+          metrics=metrics, metric_logger=metric_logger,
+          batch_based=batch_based, epoch_based=epoch_based)
+      callbacks.append(logging_metric_callback)
+
+  return callbacks
+
+
+def generate_synthetic_input_dataset(image_size, batch_size):
+  """Generate synthetic dataset."""
+  input_shape = (batch_size,) + image_size + (_NUM_CHANNELS,)
+
+  images = tf.zeros(input_shape, dtype=tf.float32)
+  labels = tf.zeros((batch_size, _NUM_CLASSES), dtype=tf.float32)
+
+  return tf.data.Dataset.from_tensors((images, labels)).repeat()
+
+
+def get_default_image_size(model):
+  """Provide default image size for each model."""
+  image_size = (224, 224)
+  if model in ["inception", "xception", "inceptionresnet"]:
+    image_size = (299, 299)
+  elif model in ["nasnetlarge"]:
+    image_size = (331, 331)
+  return image_size
+
+
+def main(_):
+  # Ensure a valid model name was supplied via command line argument
+  if FLAGS.model not in MODELS.keys():
+    raise AssertionError("The --model command line argument should "
+                         "be a key in the `MODELS` dictionary.")
+
+  # Load the model
+  tf.logging.info("Benchmark on {} model...".format(FLAGS.model))
+  keras_model = MODELS[FLAGS.model]
+  model = keras_model(weights=None)
+
+  image_size = get_default_image_size(FLAGS.model)
+  if FLAGS.use_synthetic_data:
+    tf.logging.info("Using synthetic dataset...")
+    batch_size = resnet_run_loop.per_device_batch_size(
+        FLAGS.batch_size, flags_core.get_num_gpus(FLAGS))
+    train_dataset = generate_synthetic_input_dataset(image_size, batch_size)
+    val_dataset = generate_synthetic_input_dataset(image_size, batch_size)
+  else:
+    # Use the actual ImageNet dataset (TODO)
+    pass
```

Comment:
Probably should add a tf.logging.warning here since we don't support real data.

Reply:
Or better yet, just raise an error with the note. Since the rest of this will fail if the appropriate vars are not defined.